break out cancel-task from delete-task

AmitRoushan opened the pull request:
Added a command cancel-task to cancel task.
Modified delete-task command to delete only completed task.
Following are scenarios handled :- 
1. delete-task can only delete task if task is COMPLETED.
2. delete-task will return if task does not exist or if task name is null.
3. cancel-task will try to cancel task and  will return error if receptor return error while cancelling task.
4. if task is already COMPLETED , cancel-task will return success but will not go for task cancellation.
5. cancel-task will return error if task does not exist or if task name is null.
[#95422348]

davidwadden commented:
Thanks!  We made some changes:

1. Cancelling a completed task returns an error now
2. task_runner_test injects a fake task_examiner

David

Commit:
break out cancel-task from delete-task

Added a command cancel-task to cancel task.
Modified delete-task command to delete only completed task.
[#95422348]

menicosia commented:
Tiny note: it's 'canceling' not 'cancelling'
